http-server/src/main/java/org/triplea/server/moderator/toolbox/api/key/validation/ApiKeyValidationServiceFactory.java

```diff
@@ -0,0 +1,47 @@
+package org.triplea.server.moderator.toolbox.api.key.validation;
+
+import java.util.concurrent.TimeUnit;
+
+import org.jdbi.v3.core.Jdbi;
+import org.triplea.lobby.server.db.ApiKeyDao;
+import org.triplea.server.http.AppConfig;
+
+import com.google.common.base.Charsets;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.hash.Hashing;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Factory class to create api key validation and security (rate-limiting) classes.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ApiKeyValidationServiceFactory {
+
+  static {
+    InvalidKeyCache.setCache(
+        CacheBuilder.newBuilder()
+            .expireAfterWrite(AppConfig.FAILED_API_KEY_CACHE_EXPIRATION, TimeUnit.MINUTES)
+            .build());
+  }
```

Comment:
Why is the Cache injected this way?
I'm concerned that this might cause weird issues when InvalidKeyCache is loaded on its own, without this class.

Reply:
Out of puritanism, I was avoiding the 'service' from knowing about the `AppConfig` class, so that all configuration is injected. I don't think that really matters when it comes to a static block like this.

Looking at this, I can see how cohesion and/or localization of code suffers, that we're getting an "effect-at-distance". You're also right that we get an odd initialization dependency.

With those considerations I moved the static block to the cache class proper: 6f9ad994c, and also marked the setter as visible for testing.